### DISCUSSION TITLE
No "Specify Jupyter Server for Connections" command

### DISCUSSION THREAD

**Discussion by maxwellreynolds** [microsoft/vscode-jupyter#13145] [Questions and Answers] [answered]
Maybe a silly question. I cannot find the option "Specify Jupyter Server for Connections" anymore on VS Code. The closest option I see is "Select Interpreter to Start Jupyter Server", but can't find an option that allows me to paste my jupyter server URL.

**Comment by DonJayamanne** [comment, marked as answer]
@maxwellreynolds 
This has changed.
If you want to connect to a remote jupyter server, then please go into the kernel picker and go into `Existing Jupyter Server ` and enter a URl in there and then select a kernel from the list
See here https://code.visualstudio.com/docs/datascience/jupyter-kernel-management

**Reply by maxwellreynolds** [reply]
Thanks!

**Reply by josalhor** [reply]
I almost turned crazy. Thanks from the future!

**Reply by david-stewart** [reply]
same here -- thanks from 2024!

**Reply by mheydasch** [reply]
It would be great if the official documentation [here](https://code.visualstudio.com/docs/datascience/notebooks-web) could be updated to reflect the changes.

**Comment by lappemic** [comment]
@DonJayamanne is there a possibility to name the kernel which i am missing? 

I have like 20 times the possibility to choose 'localhost'.

**Reply by DonJayamanne** [reply]
I'm not following you, can you share a screenshot/video recording of the issue
What do you mean by 20 times?

**Comment by lappemic** [comment]
Sorry, this was too unspecific.

I start the jupyter kernerl with "jupyter lab", then click on 'Select Kernel' > 'Existing Jupyter Server...' i get the full list of names i could choose from but are not existing kernels anymore:
![Screenshot 2024-03-05 at 11 59 10](https://github.com/microsoft/vscode-jupyter/assets/61876623/212533e6-24a8-4320-9633-02d05eef87e5)

**Reply by DonJayamanne** [reply]
You can remove items that are no longer valid by clicking on the `x` against each server.
Just hover over each server in the list, and a `x` appears on the right hand side of the item allowing you to remove invalid itmes.
this way you can remove servers that are no longer valid, before adding a new server.

**Reply by lappemic** [reply]
Ok, thanks Don.

I thought there might be a possibility that you can make your server having always the same name. But this works as well.

**Comment by hiraksarkar** [comment]
Has this changed again ? Can't find Jupyter: Existing Jupyter Server, making https://code.visualstudio.com/docs/datascience/jupyter-kernel-management#_existing-jupyter-server help page useless.

**Comment by harrystuart** [comment]
Bumping this issue. Per [https://code.visualstudio.com/docs/datascience/notebooks-web#_connect-to-a-remote-jupyter-server](https://code.visualstudio.com/docs/datascience/notebooks-web#_connect-to-a-remote-jupyter-server), does this elusive **Jupyter: Specify Jupyter Server for Connections** command actually exist? I want to connect to a remote server. The remote server has my notebook files so i can't just open a notebook to toggle between kernels as the notebooks live on the external server.

**Reply by mrharrison007** [reply]
Having same issue, not fun wrestling with the VS Code commands.

**Comment by dlcodns** [comment]
Hi, I found a solution! Try using the ``` Notebook: Select Notebook Kernel ```.

I posted it in detail on my blog, Google translate it and come see it.
If you have any questions, leave a comment plz.
[My Blog Post](https://rb-cloud.tistory.com/3)

**Reply by TT5H** [reply]
This worked. Thanks!
